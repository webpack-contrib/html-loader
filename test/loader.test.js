/* eslint-disable */
import loader from "../src";

import webpack from './webpack';
import { stats } from './helpers';

describe("HTML Loader", () => {
  test('Should process HTML with defaults', () => {
    const config = {};

    return webpack('index.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  test('Should process HTML with options.root', () => {
    const config = {
      loader: {
        root: 'fixtures'
      }
    };

    return webpack('root.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  test('Should process HTML with options.attrs', () => {
    const config = {
      loader: {
        attrs: [ 'div:data-attrs', ':custom-attrs' ]
      }
    };

    return webpack('attrs.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  test('Should process HTML with options.interpolate {Boolean}', () => {
    const config = { loader: { interpolate: true } };

    return webpack('interpolate.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => console.log(err))
  })

  test('Should process HTML with options.interpolate {String}', () => {
    const config = {
      loader: {
        interpolate: 'require'
      }
    };

    return webpack('interpolate.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  test('Should process HTML with options.minimize {Boolean}', () => {
    const config = { loader: { minimize: true } };

    return webpack('index.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  test('Should process HTML with options.minimize {Object}', () => {
    const config = {
      loader: {
        minimize: {
          removeComments: false
        }
      }
    };

    return webpack('index.js', config)
      .then((result) => stats(result))
      .then(({ loaders }) => {
        // if (loaders.err) throw new Error(loader.err)
        expect(loaders.src).toMatchSnapshot()
      })
      .catch((err) => err)
  })

  // TODO refactor
  test.skip("Should convert to requires", () => {
    const html = 'Text <img src="image.png"><img src="~bootstrap-img"> Text';
    const result = loader.call({}, html);

    expect(result).toEqual(
      'export default "Text <img src=\\"" + require("./image.png") + "\\"><img src=\\"" + require("bootstrap-img") + "\\"> Text";'
    );
    expect(result).toMatchSnapshot();
  });

  // https://gtest.skiphub.com/webpack/webpack/issues/752
  test("Should not remove attributes by default", () => {
    const result = loader.call({ minimize: true }, '<input type="text" />')

    expect(result).toEqual(
      'export default "<input type=text />";'
    );
    expect(result).toMatchSnapshot();
  });


  test.skip("Should not translate root-relative urls (without root query)", () => {
    const result = loader.call({}, 'Text <img src="/image.png">')

    expect(result).toEqual(
      'export default "Text <img src=\\"/image.png\\">";'
    );
    expect(result).toMatchSnapshot();
  });

  test.skip("Should ignore hash fragments in URLs", () => {
    const result = loader.call({}, '<img src="icons.svg#hash">')

    expect(result).toEqual(
      'export default "<img src=\\"" + require("./icons.svg") + "#hash\\">";'
    );
    expect(result).toMatchSnapshot();
  });

  test("Should export as ES2015 Module", () => {
    const result = loader.call({ query: {} }, '<p>Hello world!</p>');

    expect(result).toEqual('export default "<p>Hello world!</p>";');
    expect(result).toMatchSnapshot();
  });
});
