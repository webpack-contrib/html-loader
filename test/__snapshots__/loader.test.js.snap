// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Loader Should export as ES2015 Module 1`] = `"export default \\"<p>Hello world!</p>\\";"`;

exports[`HTML Loader Should not remove attributes by default 1`] = `"export default \\"<input type=text />\\";"`;

exports[`HTML Loader Should process HTML with defaults 1`] = `"export default \\"<!DOCTYPE html>\\\\n<html lang=\\\\\\"en\\\\\\">\\\\n<head>\\\\n  <meta charset=\\\\\\"utf-8\\\\\\">\\\\n  <title>HTML Loader</title>\\\\n</head>\\\\n<body>\\\\n  <!-- I'm a comment -->\\\\n  <div id=\\\\\\"app\\\\\\"></div>\\\\n</body>\\\\n</html>\\\\n\\";"`;

exports[`HTML Loader Should process HTML with options.attrs 1`] = `"throw new Error(\\"Module parse failed: /Users/Cini/Github/Webpack/webpack-loaders/html-loader/src/cjs.js??ref--0-0!/Users/Cini/Github/Webpack/webpack-loaders/html-loader/test/fixtures/attrs.html Unexpected token (1:33)\\\\nYou may need an appropriate loader to handle this file type.\\\\n| export default \\\\\\"<div data-attrs=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"foo\\\\\\\\\\\\\\"\\\\\\"')\\\\\\"></div>\\\\\\\\n\\\\\\\\n<custom-element custom-attrs=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"foo\\\\\\\\\\\\\\"\\\\\\"')\\\\\\"></custom-element>\\\\\\\\n<custom-container custom-attrs=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"foo\\\\\\\\\\\\\\"\\\\\\"')\\\\\\"></custom-container>\\\\\\\\n\\\\\\";\\");"`;

exports[`HTML Loader Should process HTML with options.interpolate {Boolean} 1`] = `"throw new Error(\\"Module parse failed: /Users/Cini/Github/Webpack/webpack-loaders/html-loader/src/cjs.js??ref--0-0!/Users/Cini/Github/Webpack/webpack-loaders/html-loader/test/fixtures/interpolate.html Unexpected token (1:15)\\\\nYou may need an appropriate loader to handle this file type.\\\\n| export default <img src=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"\${img.src}\\\\\\\\\\\\\\"\\\\\\"')\\\\\\" alt=\\\\\\"\${img.alt}\\\\\\">\\\\n| <img src=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"\${require(./image.png)}\\\\\\\\\\\\\\"\\\\\\"')\\\\\\">\\\\n| ;\\");"`;

exports[`HTML Loader Should process HTML with options.interpolate {String} 1`] = `"throw new Error(\\"Module parse failed: /Users/Cini/Github/Webpack/webpack-loaders/html-loader/src/cjs.js??ref--0-0!/Users/Cini/Github/Webpack/webpack-loaders/html-loader/test/fixtures/interpolate.html Unexpected token (1:26)\\\\nYou may need an appropriate loader to handle this file type.\\\\n| export default \\\\\\"<img src=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"\${img.src}\\\\\\\\\\\\\\"\\\\\\"')\\\\\\" alt=\\\\\\\\\\\\\\"\${img.alt}\\\\\\\\\\\\\\">\\\\\\\\n<img src=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"\${require(./image.png)}\\\\\\\\\\\\\\"\\\\\\"')\\\\\\">\\\\\\\\n\\\\\\";\\");"`;

exports[`HTML Loader Should process HTML with options.minimize {Boolean} 1`] = `"export default \\"<!DOCTYPE html> <html lang=en> <head> <meta charset=utf-8> <title>HTML Loader</title> </head> <body> <div id=app></div> </body> </html> \\";"`;

exports[`HTML Loader Should process HTML with options.minimize {Object} 1`] = `"export default \\"<!DOCTYPE html> <html lang=en> <head> <meta charset=utf-8> <title>HTML Loader</title> </head> <body> <!-- I'm a comment --> <div id=app></div> </body> </html> \\";"`;

exports[`HTML Loader Should process HTML with options.root 1`] = `"throw new Error(\\"Module parse failed: /Users/Cini/Github/Webpack/webpack-loaders/html-loader/src/cjs.js??ref--0-0!/Users/Cini/Github/Webpack/webpack-loaders/html-loader/test/fixtures/root.html Unexpected token (1:26)\\\\nYou may need an appropriate loader to handle this file type.\\\\n| export default \\\\\\"<img src=\\\\\\"require('\\\\\\"./\\\\\\\\\\\\\\"./image.png\\\\\\\\\\\\\\"\\\\\\"')\\\\\\">\\\\\\\\n\\\\\\";\\");"`;
